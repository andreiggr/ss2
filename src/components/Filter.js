import React from "react";

const Filter = ({ title, options }) => {
  return (
    <div
      style={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center"
      }}
    >
      <p style={{ marginRight: "15px" }}>{title}</p>
      <select>
        {options.map((title, i) => (
          <option key={i} value={title}>
            {title}
          </option>
        ))}
      </select>
    </div>
  );
};

export default Filter;
