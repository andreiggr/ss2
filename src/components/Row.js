import React from "react";

const Row = ({ row }) => {
  return (
    <div style={{display:"flex"}}>
      {row.map((item, i) => (
        <div
          style={{
            width: "150px",
            overflowWrap: "break-word",
            wordWrap: "break-word",
            border: "1px solid black",
            padding: "5px 5px 25px 5px"
          }}
          key={i}
        >
          <p>{item}</p>
        </div>
      ))}
    </div>
  );
};

export default Row;
