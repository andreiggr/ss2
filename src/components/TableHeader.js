import React from "react";

const TableHeader = ({ headerData }) => {
  return (
    <div style={{ display: "flex" }}>
      {headerData &&
        headerData.map((item, i) => (
          <div
            style={{
              width: "150px",
              overflowWrap: "break-word",
              wordWrap: "break-word",
              border: "1px solid black",
              padding: "5px 5px 25px 5px",
              background: "#EEEFF5"
              
            }}
            key={i}
          >
            <p style={{fontWeight:"bold"}}>{item.$t}</p>
          </div>
        ))}
    </div>
  );
};

export default TableHeader;
