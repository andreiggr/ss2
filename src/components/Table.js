import React from "react";
import Row from "./Row";
import TableHeader from "./TableHeader";
import { connect } from "react-redux";
import useFilter from "../utils/useFilter";

export const Table = ({ items, filter }) => {
  //select data for table header
  const tableHeader = items[0];

  //select data for the main table with filters applied
  const tableData = [...items.slice(1)];
  const tableFiltered = useFilter(tableData, filter)

  return (
    <div style={{ display: "flex", flexDirection: "column" }}>
      <TableHeader headerData={tableHeader} />
      {tableFiltered.map((row, index) => (
        <Row key={index} row={row} />
      ))}
      <p> Total results: {tableFiltered.length} </p>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    items: state.items,
    filter: state.filter
  };
};

export default connect(mapStateToProps)(Table);
