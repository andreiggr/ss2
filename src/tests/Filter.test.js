import React from "react";
import ReactDOM from "react-dom";
import { Filter } from "../components/Filter";
import { shallow } from "enzyme";

describe("Filter Render Test", () => {
  it("displays title and options ok", () => {
    const title = "mockTitle";
    const appComponent = shallow(<Filter title={title} />);

    expect(appComponent.getElements()).toMatchSnapshot();
  });
});
