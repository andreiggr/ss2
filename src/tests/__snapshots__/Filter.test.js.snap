exports[`Filter Render Test displays title and options ok 1`] = `
Array [
  <div
    style={
        Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
    }>
    <p
        style={
            Object {
                "marginRight": "15px",
              }
        }>
        mockTitle
    </p>
    <select
        onChange={[Function]}>
        <option
            value="All">
            All
        </option>
        <option
            value="Vertical">
            Vertical
        </option>
        <option
            value="Horizontal">
            Horizontal
        </option>
        <option
            value="Supplementary">
            Supplementary
        </option>
    </select>
</div>,
]
`;
